### ISSUE TITLE
Estimate a reasonable nstp based on stresses

### ISSUE THREAD

**Issue by bdestombe** [ArtesiaWater/nlmod#82]
I have a routine that I'd like to include. When working with actual timeseries of, for example, recharge, I find myself always fiddling with the number of time steps per stress period for transient stress periods. 
It is straight forward to assign many nstp's to days with large rainfall, but then the model often fails to converge on the day after the intense rainfall. I wrote the following piece of code to help me with that. This code ensures there are also nstp's assigned to the days after the rainfall event. An appoach is applied that honors tsmult.

Please, consider this issue as a placeholder for a proper pull request.

```python
def estimate_nstp(forcing, perlen=1, tsmult=1.1, nstp_min=1, nstp_max=25):
    """ Scale the nstp's linearly between the min and max of the forcing.

    Ensures that the first time step of this stress period connects to the last time step of the previous stress period.
    The ratio between the two time-step durations can be at most tsmult."""

    nt = len(forcing)

    # Scaled linear between min and max. array nstp will be modified along the way
    nstp = np.round((forcing - forcing.min(axis=0)) /
                   (forcing.max(axis=0) - forcing.min(axis=0)) * (nstp_max - nstp_min)
                   + nstp_min).astype(int)
    perlen = np.full(nt, fill_value=perlen)
    tsmult = np.full(nt, fill_value=tsmult)

    # Duration of the first time step of each stress period. Equation TM6A16 p.4-5 eq.1
    dt0_arr = perlen * (tsmult - 1) / (tsmult ** nstp - 1)
    # Add axis so that durations of subsequent time steps can be added for each stress period.
    dt_arr = [[i] for i in dt0_arr]

    for i, (perleni, tsmulti, nstpi, dt0) in enumerate(zip(perlen, tsmult, nstp, dt0_arr)):
        for _ in range(nstpi - 1):
            dt_i = dt_arr[i][-1] * tsmulti
            dt_arr[i].append(dt_i)

        # reduce the length of the first time step of the next period to match the last time step of this period.
        dt0_next = dt_arr[i][-1] * tsmulti
        if i <= nt - 2 and dt0_next < dt_arr[i + 1][0]:
            # Equation derived from TM6A16 p.4-5 eq.1
            nstp_next = np.ceil(np.log(perleni * (tsmulti - 1) / dt0_next + 1) / np.log(tsmulti)).astype(int)
            nstp[i + 1] = nstp_next
            
            # Equation TM6A16 p.4-5 eq.1
            dt_arr[i + 1] = [perlen[i + 1] * (tsmult[i + 1] - 1) / (tsmult[i + 1] ** nstp_next - 1)]

    nstp = [len(i) for i in dt_arr]
    return nstp
```
The code should also accept multiple stress time series. Not sure if it is useful, but log-scaling between min and max is also an easy option to implement. Note to self: check if it should be `np.ceil` or `np.round`.

**Comment by bdestombe** [comment]
Add an option to return also `dt_arr`. Use `dt_arr` to test the function with various perlen and tsmult.

**Comment by bdestombe** [comment]
The linear scaling should use the absolute values

**Comment by bdestombe** [comment]
nstp is defined in the loop and again at the end. Both result in the same values, thus the last line is not needed.

**Comment by dbrakenhoff** [comment]
Closed by #99
